test/unit/org/apache/cassandra/metrics/CounterMutationMetricsTest.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    @Test
+    public void testLocksDuplicate()
+    {
+        AssertMetrics metrics = new AssertMetrics();
+
+        ColumnFamilyStore cfsOne = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        ColumnFamilyStore cfsTwo = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF2);
+
+        cfsOne.truncateBlocking();
+        cfsTwo.truncateBlocking();
+
+        Mutation.PartitionUpdateCollector batch = new Mutation.PartitionUpdateCollector(KEYSPACE1, Util.dk("key1"));
+        batch.add(new RowUpdateBuilder(cfsOne.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 1L)
+                  .add("val2", -1L)
+                  .build().getPartitionUpdate(cfsOne.metadata()));
+
+        batch.add(new RowUpdateBuilder(cfsTwo.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 2L)
+                  .add("val2", -2L)
+                  .build().getPartitionUpdate(cfsTwo.metadata()));
+
+        new CounterMutation(batch.build(), ConsistencyLevel.ONE).apply();
+
+        metrics.assertLockTimeout(0);
+        metrics.assertLocksPerUpdate(1);
+    }
+
+    @Test
+    public void testLockTimeout() throws Throwable
+    {
+        AssertMetrics metrics = new AssertMetrics(11L);
+
+        ColumnFamilyStore cfs = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        cfs.truncateBlocking();
+
+        // Obtain lock
+        new CounterMutation(
+        new RowUpdateBuilder(cfs.metadata(), 5, "key1").clustering("cc")
+                                                       .add("val", 1L)
+                                                       .build(),
+        ConsistencyLevel.ONE).grabCounterLocks(Keyspace.open(KEYSPACE1), new ArrayList<>());
+
+        metrics.assertLockTimeout(0);
+
+        // Try to mutate
+        CompletableFuture.runAsync(() ->
+                                   {
+                                       try
+                                       {
+                                           new CounterMutation(
+                                           new RowUpdateBuilder(cfs.metadata(), 5, "key1")
+                                           .clustering("cc")
+                                           .add("val", 1L)
+                                           .add("val2", -1L)
+                                           .build(),
+                                           ConsistencyLevel.ONE).apply();
+                                       }
+                                       catch (WriteTimeoutException e)
+                                       {
+                                           // Expected exception.
+                                           // If not met, lockTimeout counter will not be increased, which is asserted.
+                                       }
+                                   }).get(100, TimeUnit.MILLISECONDS);
```

Comment:
I'd rather we had a constant named `longerThanCounterWriteRpcTimeout`, with an appropriate assertion
again - when the test fails I'd like to be as sure as possible that no race condition happened

btw, could we make it more like 5s? Or 15s? IIUC when the test passes we don't wait for such a long time anyway, so I'd rather have 5s than have to verify if some VM got stuck during testing should the test fail in CI.